pkg/apis/v1/nodeclaim_validation.go

```diff
@@ -0,0 +1,285 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/samber/lo"
+	"go.uber.org/multierr"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/ptr"
+)
+
+var (
+	SupportedNodeSelectorOps = sets.NewString(
+		string(v1.NodeSelectorOpIn),
+		string(v1.NodeSelectorOpNotIn),
+		string(v1.NodeSelectorOpGt),
+		string(v1.NodeSelectorOpLt),
+		string(v1.NodeSelectorOpExists),
+		string(v1.NodeSelectorOpDoesNotExist),
+	)
+
+	SupportedReservedResources = sets.NewString(
+		v1.ResourceCPU.String(),
+		v1.ResourceMemory.String(),
+		v1.ResourceEphemeralStorage.String(),
+		"pid",
+	)
+
+	SupportedEvictionSignals = sets.NewString(
+		"memory.available",
+		"nodefs.available",
+		"nodefs.inodesFree",
+		"imagefs.available",
+		"imagefs.inodesFree",
+		"pid.available",
+	)
+)
+
+func (in *NodeClaim) SupportedVerbs() []admissionregistrationv1.OperationType {
```

Comment:
Can this be removed for the same reason? We shouldn't be supporting the webhooks anymore

Reply:
This PR is meant to grab the duplicate the v1beta1 APIs then create PR to remove the addition fields later